### PR TITLE
Update system form webUI

### PR DESCRIPTION
**Pull request by mariogrip**
This will download form branch "update", it checking for updates from a webserver. (I can send you the file so you can have them on minepeon's server) So when we push a update to branch "update" and edit the update.xml file, you will get a message on minepeon's webUI. 

image

also i added so if the mh/s is bigger then 1000 it will convert it to gh/s

### CODE REVIEW

**Comment by MineForeman** [comment]
This update is huge and is going to take me a while to test, it also addresses a few different things (ideally commits should be for just one thing).

I like the update system a lot, but it totally changes the way people update their systems and how I package the system.  I have to be sure on all of the implications of it.

Neil

**Comment by mariogrip** [comment]
Ok, so how the update system work. it checks every hour for a update. the if there is, there will be a notification under "navbar". when the user click on "Do you wanna update?" it will download the  updates form a branch named update.

What we need to do for pushing update
1. Get the new commits for master to update

```
git checkout update
git pull origin master
```
1. Update the xml file

```
<?xml version="1.0" encoding="utf-8"?>
<update>
<version>MinePeon 0.2.4.5</version>
<branch>update</branch>
</update>
```

**Comment by MineForeman** [comment]
It sounds simple enough, but it is still a huge architectural change and I need to go thorough it thoroughly.  Does it break git pull?  What are the security concerns?

To add to that apache 2.4 breaks mod_php and running php using fast cgi (the recommended alternative) breaks php_value settings in htaccess (required for allow_url_fopen) so that needs to be taken over to php.ini once I finish re-doing it for apache 2.4.  Then I need to test all that!!!!

It is all good stuff and doable but it blows the chance of an image update today ;) .  When we talk on skype we will work out things so they are not a total surprise to me!

Neil

**Comment by mariogrip** [comment]
Does it break git pull: It does not, it uses git. (it does the same as "git pull origin update")
What are the security concerns: it uses git! same as git pull (git fetch and git merge).

**Comment by mariogrip** [comment]
The code is a mess because i wanted live progress on front end while back end is running...

**Comment by MineForeman** [comment]
Sorry, I have been distracted by the apache issues (and I am at work).  I had a brief look but I saw variables like;-

output1
outputa
outputb

And decided it would be better to wait until after work.

**Comment by mariogrip** [comment]
Outputs error if there is any.

this just make from array to string.

```
foreach ($outputa as $outputb) {
     $output = $output . $outputb;
}
```

**Comment by MineForeman** [comment]
It's just a bit hard to follow.  I will have a look and pull when I get home.

Neil

**Comment by mariogrip** [comment]
I will get to sleep, i will be back to talk with you later. like 07:00 - 09:00 CET

btw, my local time: http://www.timeanddate.com/worldclock/city.html?n=187

**Comment by MineForeman** [comment]
Great stuff!  Pulling now.

I am going to call a feature freeze until I can get the apache stuff sorted and get a release out (with this).  Keep doing bug's as you find em.
